### PR TITLE
Feature/google oauth

### PR DESCRIPTION
**Pull request by kamicut**
Adds google based authentication:
* Changes models to use `userId`and `email` in tokens
* Adds an environment variable to determine which authentication scheme to use
* Adds google `bell` strategy

TODO:
- [x] Change all instances of `osmId`to `userId`
- [x] Change tests

### CODE REVIEW

**Comment by kamicut** [comment]
@vgeorge there is a field `isAdmin` which I'm not sure at what point in the login flow it has to be filled? Does it have to be in the credentials?

**Comment by kamicut** [comment]
This PR is ready for review! @vgeorge I'd like to know if you'd like me to add any extra tests?

**Review by vgeorge — CHANGES_REQUESTED** [review]
This is looking good, @kamicut. My comments:

1) Tests passed to me, but when I tried to migrate an already populated dev db instance I got this:

```
migration file "20201004185959_user-id.js" failed
migration failed with error: alter table "observations" add constraint "observations_userid_foreign" foreign key ("userId") references "users" ("id") - insert or update on table "observations" violates foreign key constraint "observations_userid_foreign"
Key (userId)=(11356778) is not present in table "users".
error: insert or update on table "observations" violates foreign key constraint "observations_userid_foreign"
    at Connection.parseE (/my_dev_dir/plasticwatch/api/node_modules/pg/lib/connection.js:604:11)
    at Connection.parseMessage (/my_dev_dir/plasticwatch/api/node_modules/pg/lib/connection.js:401:19)
    at Socket.<anonymous> (/my_dev_dir/plasticwatch/api/node_modules/pg/lib/connection.js:121:22)
    at Socket.emit (events.js:198:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
From previous event:
    at Client_PG._query (/my_dev_dir/plasticwatch/api/node_modules/knex/lib/dialects/postgres/index.js:234:12)
    at Client_PG.query (/my_dev_dir/plasticwatch/api/node_modules/knex/lib/client.js:158:17)
    at /my_dev_dir/plasticwatch/api/node_modules/knex/lib/transaction.js:331:24
From previous event:
    at Client_PG.trxClient.query (/my_dev_dir/plasticwatch/api/node_modules/knex/lib/transaction.js:326:12)
    at Runner.query (/my_dev_dir/plasticwatch/api/node_modules/knex/lib/runner.js:135:36)
    at Runner.queryArray (/my_dev_dir/plasticwatch/api/node_modules/knex/lib/runner.js:224:14)
    at /my_dev_dir/plasticwatch/api/node_modules/knex/lib/runner.js:37:25
From previous event:
    at Runner.run (/my_dev_dir/plasticwatch/api/node_modules/knex/lib/runner.js:25:16)
    at SchemaBuilder.Target.then (/my_dev_dir/plasticwatch/api/node_modules/knex/lib/interface.js:14:43)
```

2) We are already using `camelCase` in query parameters and database names, I would suggest using `createdAt` instead of `created_at` to keep consistency.

3) About the tests, they seem to cover pretty well the changes in `User` model, but ideally we need integration testing for auth as it heavily depends on the frontend. I'm not sure what is the best approach to accomplish this, but I think we might need to add cypress to Observe/PW front-end to ensure auth works for both providers.

**Comment by LanesGood** [comment]
@kamicut I was not able to successfully run this branch. 

`npm run start-dev-db` succeeds, but all other commands fail. My error messages below:

<details>
    <summary>`npm run migrate-dev-db` - Click to expand error</summary>

```bash
~/github/wikiplastic-api $ npm run migrate-dev-db

> observe-api@0.1.2 migrate-dev-db /Users/lane/Github/wikiplastic-api
> NODE_ENV=development node_modules/.bin/knex migrate:latest

Error: The migration directory is corrupt, the following files are missing: 20200929162606_osm-objects-ts-vector.js
    at validateMigrationList (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/migrate/Migrator.js:550:11)
    at migrationListResolver.listAllAndCompleted.then (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/migrate/Migrator.js:68:9)
From previous event:
    at Migrator.latest (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/migrate/Migrator.js:67:8)
    at Command.commander.command.description.option.action (/Users/lane/Github/wikiplastic-api/node_modules/knex/bin/cli.js:179:18)
    at Command.listener (/Users/lane/Github/wikiplastic-api/node_modules/commander/index.js:291:8)
    at Command.emit (events.js:198:13)
    at Command.parseArgs (/Users/lane/Github/wikiplastic-api/node_modules/commander/index.js:672:12)
    at Command.parse (/Users/lane/Github/wikiplastic-api/node_modules/commander/index.js:459:21)
    at Liftoff.invoke (/Users/lane/Github/wikiplastic-api/node_modules/knex/bin/cli.js:332:13)
    at Liftoff.execute (/Users/lane/Github/wikiplastic-api/node_modules/liftoff/index.js:201:12)
    at module.exports (/Users/lane/Github/wikiplastic-api/node_modules/flagged-respawn/index.js:51:3)
    at Liftoff.<anonymous> (/Users/lane/Github/wikiplastic-api/node_modules/liftoff/index.js:191:5)
    at /Users/lane/Github/wikiplastic-api/node_modules/liftoff/index.js:149:9
    at /Users/lane/Github/wikiplastic-api/node_modules/v8flags/index.js:121:14
    at FSReqWrap.args [as oncomplete] (fs.js:140:20)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! observe-api@0.1.2 migrate-dev-db: `NODE_ENV=development node_modules/.bin/knex migrate:latest`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the observe-api@0.1.2 migrate-dev-db script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lane/.npm/_logs/2020-10-20T17_46_25_228Z-debug.log
```

</details>

<details>
    <summary>`npm run seed-dev` - Click to expand error</summary>

```bash
~/github/wikiplastic-api $ npm run seed-dev

> observe-api@0.1.2 seed-dev /Users/lane/Github/wikiplastic-api
> NODE_ENV=development node_modules/.bin/knex seed:run

Seeding users...
Error: Error while executing "/Users/lane/Github/wikiplastic-api/seeds/1-users.js" seed: insert into "users" ("displayName", "email", "isAdmin", "osmCreatedAt", "osmDisplayName", "osmId") values ($1, $2, $3, $4, $5, $6) returning * - column "displayName" of relation "users" does not exist
    at Promise.resolve.then.then.catch (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/seed/Seeder.js:146:25)
    at Connection.parseE (/Users/lane/Github/wikiplastic-api/node_modules/pg/lib/connection.js:604:11)
    at Connection.parseMessage (/Users/lane/Github/wikiplastic-api/node_modules/pg/lib/connection.js:401:19)
    at Socket.<anonymous> (/Users/lane/Github/wikiplastic-api/node_modules/pg/lib/connection.js:121:22)
    at Socket.emit (events.js:198:13)
    at Socket.EventEmitter.emit (domain.js:448:20)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
    at Client_PG._query (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/dialects/postgres/index.js:234:12)
    at Client_PG.query (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/client.js:158:17)
    at Runner.query (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/runner.js:135:36)
    at /Users/lane/Github/wikiplastic-api/node_modules/knex/lib/runner.js:39:23
    at Runner.run (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/runner.js:25:16)
    at Builder.Target.then (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/interface.js:14:43)
    at process._tickCallback (internal/process/next_tick.js:68:7)
From previous event:
    at each (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/seed/Seeder.js:140:23)
    at arrayEach (/Users/lane/Github/wikiplastic-api/node_modules/lodash/lodash.js:516:11)
    at forEach (/Users/lane/Github/wikiplastic-api/node_modules/lodash/lodash.js:9342:14)
    at Seeder._waterfallBatch (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/seed/Seeder.js:135:3)
    at Seeder.<anonymous> (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/seed/Seeder.js:90:21)
From previous event:
    at Seeder.<anonymous> (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/seed/Seeder.js:89:34)
    at runCallback (timers.js:705:18)
    at tryOnImmediate (timers.js:676:5)
    at processImmediate (timers.js:658:5)
From previous event:
    at Seeder._runSeeds (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/seed/Seeder.js:88:6)
    at Seeder._seedData.bind.then (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/seed/Seeder.js:37:19)
    at runCallback (timers.js:705:18)
    at tryOnImmediate (timers.js:676:5)
From previous event:
    at Seeder.run (/Users/lane/Github/wikiplastic-api/node_modules/knex/lib/seed/Seeder.js:31:6)
    at Command.commander.command.description.option.option.action (/Users/lane/Github/wikiplastic-api/node_modules/knex/bin/cli.js:319:15)
    at Command.listener (/Users/lane/Github/wikiplastic-api/node_modules/commander/index.js:291:8)
    at Command.emit (events.js:198:13)
    at Command.parseArgs (/Users/lane/Github/wikiplastic-api/node_modules/commander/index.js:672:12)
    at Command.parse (/Users/lane/Github/wikiplastic-api/node_modules/commander/index.js:459:21)
    at Liftoff.invoke (/Users/lane/Github/wikiplastic-api/node_modules/knex/bin/cli.js:332:13)
    at Liftoff.execute (/Users/lane/Github/wikiplastic-api/node_modules/liftoff/index.js:201:12)
    at module.exports (/Users/lane/Github/wikiplastic-api/node_modules/flagged-respawn/index.js:51:3)
    at Liftoff.<anonymous> (/Users/lane/Github/wikiplastic-api/node_modules/liftoff/index.js:191:5)
    at /Users/lane/Github/wikiplastic-api/node_modules/liftoff/index.js:149:9
    at /Users/lane/Github/wikiplastic-api/node_modules/v8flags/index.js:138:14
    at /Users/lane/Github/wikiplastic-api/node_modules/v8flags/index.js:41:14
    at /Users/lane/Github/wikiplastic-api/node_modules/v8flags/index.js:53:7
    at process._tickCallback (internal/process/next_tick.js:61:11)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! observe-api@0.1.2 seed-dev: `NODE_ENV=development node_modules/.bin/knex seed:run`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the observe-api@0.1.2 seed-dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/lane/.npm/_logs/2020-10-20T17_46_32_020Z-debug.log
```

</details>

<details>
  <summary>npm run dev - click to expand error</summary>

```bash
~/github/wikiplastic-api $ npm run dev

> observe-api@0.1.2 dev /Users/lane/Github/wikiplastic-api
> NODE_ENV=development node_modules/.bin/nodemon index.js

[nodemon] 1.19.2
[nodemon] to restart at any time, enter `rs`
[nodemon] watching dir(s): *.*
[nodemon] starting `node index.js`
/Users/lane/Github/wikiplastic-api/node_modules/config/lib/config.js:1
Error: Configuration property "googleOAuth" is not defined
    at Proxy.Config.get (/Users/lane/Github/wikiplastic-api/node_modules/config/lib/config.js:182:11)
    at Object.<anonymous> (/Users/lane/Github/wikiplastic-api/app/services/auth/google.js:8:12)
    at Generator.next (<anonymous>)
    at Object.<anonymous> (/Users/lane/Github/wikiplastic-api/app/services/auth/oauth.js:1)
    at Generator.next (<anonymous>)
    at Object.<anonymous> (/Users/lane/Github/wikiplastic-api/app/services/auth/index.js:1)
[nodemon] app crashed - waiting for file changes before starting...
```

</details>

And back in the start-dev-db terminal I see:

<details>
    <summary>`npm run start-dev-db` - Click to expand output</summary>

```bash
~/github/wikiplastic-api $ npm run start-dev-db

> observe-api@0.1.2 start-dev-db /Users/lane/Github/wikiplastic-api
> docker-compose -f docker-compose-dev.yml up

Starting wikiplastic-api_postgresql-observe-dev_1 ... done
Attaching to wikiplastic-api_postgresql-observe-dev_1
postgresql-observe-dev_1  | 2020-10-20 17:44:58.757 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgresql-observe-dev_1  | 2020-10-20 17:44:58.757 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgresql-observe-dev_1  | 2020-10-20 17:44:58.789 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgresql-observe-dev_1  | 2020-10-20 17:44:59.017 UTC [20] LOG:  database system was interrupted; last known up at 2020-10-20 17:42:10 UTC
postgresql-observe-dev_1  | 2020-10-20 17:45:07.952 UTC [20] LOG:  database system was not properly shut down; automatic recovery in progress
postgresql-observe-dev_1  | 2020-10-20 17:45:08.113 UTC [20] LOG:  redo starts at 0/30E3DE0
postgresql-observe-dev_1  | 2020-10-20 17:45:08.114 UTC [20] LOG:  invalid record length at 0/30E3E18: wanted 24, got 0
postgresql-observe-dev_1  | 2020-10-20 17:45:08.115 UTC [20] LOG:  redo done at 0/30E3DE0
postgresql-observe-dev_1  | 2020-10-20 17:45:08.398 UTC [1] LOG:  database system is ready to accept connections
postgresql-observe-dev_1  | 2020-10-20 17:46:32.017 UTC [28] ERROR:  column "displayName" of relation "users" does not exist at character 22
postgresql-observe-dev_1  | 2020-10-20 17:46:32.017 UTC [28] STATEMENT:  insert into "users" ("displayName", "email", "isAdmin", "osmCreatedAt", "osmDisplayName", "osmId") values ($1, $2, $3, $4, $5, $6) returning *
```

</details>

File: app/services/auth/index.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import setupOAuth from './oauth';
 import { setupJwtAuth } from './jwt';
 
+const config = require('config');
+
 async function setupAuth (server) {
-  await setupOAuth(server);
+  await setupOAuth(server, config.get('oauthStrategy'));
```
Comment (geohacker) [inline app/services/auth/index.js:7]: So is this required anymore?

**Review by geohacker — APPROVED** [review]

**Pull request merged**